classes/plugins/PKPPubIdPlugin.inc.php

```diff
@@ -0,0 +1,410 @@
+	 * @param $preview boolean
+	 *  when true, the public identifier will not be stored
+	 * @return string
+	 */
+	abstract function getPubId($pubObject, $preview = false);
+
+	/**
+	 * Construct the public identifier from its prefix and suffix.
+	 * @param $pubIdPrefix string
+	 * @param $pubIdSuffix string
+	 * @param $contextId integer
+	 * @return string
+	 */
+	abstract function constructPubId($pubIdPrefix, $pubIdSuffix, $contextId);
+
+	/**
+	 * Public identifier type, see
+	 * http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html
+	 * @return string
+	 */
+	abstract function getPubIdType();
+
+	/**
+	 * Public identifier type that will be displayed to the reader.
+	 * @return string
+	 */
+	abstract function getPubIdDisplayType();
+
+	/**
+	 * Full name of the public identifier.
+	 * @return string
+	 */
+	abstract function getPubIdFullName();
+
+	/**
+	 * Get the whole resolving URL.
+	 * @param $contextId integer
+	 * @param $pubId string
+	 * @return string resolving URL
+	 */
+	abstract function getResolvingURL($contextId, $pubId);
+
+	/**
+	 * Get the file (path + filename)
+	 * to be included into the object's
+	 * metadata pages.
+	 * @return string
+	 */
+	abstract function getPubIdMetadataFile();
+
+	/**
+	 * Get the class name of the settings form.
+	 * @return string
+	 */
+	abstract function getSettingsFormName();
+
+	/**
+	 * Get the additional form field names.
+	 * @return array
+	 */
+	abstract function getFormFieldNames();
+
+	/**
+	 * Get the the prefix form field name.
+	 * @return string
+	 */
+	abstract function getPrefixFieldName();
+
+	/**
+	 * Get the the suffix form field name.
+	 * @return string
+	 */
+	abstract function getSuffixFieldName();
+
+	/**
+	 * Get the the suffix patterns form field names.
+	 * @return array (pub object type => suffix pattern field name)
+	 */
+	abstract function getSuffixPatternsFieldNames();
+
+	/**
+	 * Get the checkbox form field name that
+	 * is used to define if a pub object should
+	 * be excluded from assigning the pub id to it.
+	 * @return string
+	 */
+	abstract function getExcludeFormFieldName();
+
+	/**
+	 * Get additional field names to be considered for storage.
+	 * @return array
+	 */
+	abstract function getDAOFieldNames();
+
+	/**
+	 * Get the possible publication object types.
+	 * @return array
+	 */
+	abstract function getPubObjectTypes();
+
+	/**
+	 * Get all publication objects of the given type.
+	 * @param $pubObjectType object
+	 * @param $contextId integer
+	 * @return array
+	 */
+	abstract function getPubObjects($pubObjectType, $contextId);
+
+	/**
+	 * Is this object type enabled in plugin settings
+	 * @param $pubObjectType object
+	 * @param $contextId integer
+	 * @return boolean
+	 */
+	abstract function isObjectTypeEnabled($pubObjectType, $contextId);
+
+	/**
+	 * Should the object be excluded from
+	 * assigning it the pub id
+	 * @param $pubObject object
+	 * @return boolean
+	 */
+	function isObjectExcluded($pubObject) {
```

Comment:
I'm hoping we can get rid of some of the exclusion stuff if we do what we discussed -- including the assignment step somewhere in the publication workflow. It'll be nice to reduce the number of abstract methods here.